libraries/chain/include/eosio/chain/fork_database.hpp

```diff
@@ -46,11 +44,11 @@ namespace eosio::chain {
       using full_branch_t    = std::vector<bhsp_t>;
       using branch_pair_t    = pair<branch_t, branch_t>;
 
-      explicit fork_database_t(uint32_t magic_number = legacy_magic_number);
+      explicit fork_database_t();
       ~fork_database_t();
 
-      void open( const std::filesystem::path& fork_db_file, validator_t& validator );
-      void close( const std::filesystem::path& fork_db_file );
+      void open( const std::filesystem::path& fork_db_file, fc::cfile_datastream& ds, validator_t& validator );
+      void close( std::ofstream& out );
```

Comment:
I'm curious the pros/cons of putting both legacy and new into the same file. I was expecting this to use two files. This means the new format will always support reading legacy info. Is there an advantage of storing in one file instead of storing two diff fork database files?

Reply:
The new code still needs to be able to read the old format regardless. 
Since we need to save additional information, and can't re-use the old code anyways, it feels more natural to increment the version and update the file format.
Probably more user friendly as well for users who might be puzzled to see two files if the stop nodeos during the transition.
Also, because we need to store which db is in use, two files using the old read/write code would not have sufficed.